spanner/row.go

```diff
@@ -378,3 +386,133 @@ func (r *Row) ToStructLenient(p interface{}) error {
 		true,
 	)
 }
+
+// SelectAll scans rows into a slice (v)
+func SelectAll(rows Iterator, v interface{}, options ...DecodeOptions) error {
+	if rows == nil {
+		return fmt.Errorf("rows is nil")
+	}
+	if v == nil {
+		return fmt.Errorf("p is nil")
+	}
+	vType := reflect.TypeOf(v)
+	if k := vType.Kind(); k != reflect.Ptr {
+		return errToStructArgType(v)
+	}
+	sliceType := vType.Elem()
+	if reflect.Slice != sliceType.Kind() {
+		return errNotASlicePointer()
+	}
+	sliceVal := reflect.Indirect(reflect.ValueOf(v))
+	itemType := sliceType.Elem()
+	s := &decodeSetting{}
+	for _, opt := range options {
+		opt.Apply(s)
+	}
+
+	isPrimitive := itemType.Kind() != reflect.Struct
+	var pointers []interface{}
+	isFistRow := true
+	rowIndex := -1
+	return rows.Do(func(row *Row) error {
+		sliceItem := reflect.New(itemType).Elem()
```

Comment:
if I'm following correctly, this implies the type of `v` must be something like `*[]*struct {...}`.

Would it be too hard to also support `*[]struct { ... }`? The benefit for users that pass a `*[]struct { ... }` would be less calls to the allocator, lower allocated memory, and better memory locality (as all rows are guaranteed to be contiguous in memory).

Reply:
Sorry, my bad, misread the code